tests/presets.spec.js

```diff
@@ -740,6 +740,445 @@ describe('Presets', function() {
+            'version = \'1.0.0\'',
+            '',
+            '[dependencies.bar]',
+            'version = \'2.0.0\'',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.cargoToml);
+          this.cwd.removeCallback();
+        });
+
+        it('should be able to update the version (and preserve the single-quotes)', function(done) {
+          presets.updateVersion.cargo({}, this.cwd.name, '1.1.0', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.cargoToml, 'utf8')).to.equal([
+              '[package]',
+              'name = \'foo\'',
+              'version = \'1.1.0\'',
+              '',
+              '[dependencies.bar]',
+              'version = \'2.0.0\'',
+              ''
+            ].join('\n'));
+
+            done();
+          });
+        });
+
+      });
+
+      describe('missing Cargo.toml and existing Cargo.lock', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.cargoLock = path.join(this.cwd.name, 'Cargo.lock');
+
+          fs.writeFileSync(this.cargoLock, [
+            '[[package]]',
+            'name = "bar"',
+            'version = "2.0.0"',
+            'source = "registry+https://github.com/rust-lang/crates.io-index"',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.cargoLock);
+          this.cwd.removeCallback();
+        });
+
+        it('should yield an error', function(done) {
+          presets.updateVersion.cargo({}, this.cwd.name, '1.0.1', (error) => {
+            m.chai.expect(error).to.be.an.instanceof(Error);
+            m.chai.expect(error.code).to.equal('ENOENT');
+            done();
+          });
+        });
+
+      });
+
+      describe('updating Cargo.toml with unchanged version', function() {
+
+        beforeEach(function() {
+          this.cwd = tmp.dirSync();
+          this.cargoToml = path.join(this.cwd.name, 'Cargo.toml');
+
+          fs.writeFileSync(this.cargoToml, [
+            '[package]',
+            'name = "foo"',
+            'version = "1.2.3"',
+            '',
+            '[dependencies.bar]',
+            'version = "2.0.0"',
+            ''
+          ].join('\n'));
+        });
+
+        afterEach(function() {
+          fs.unlinkSync(this.cargoToml);
+          this.cwd.removeCallback();
+        });
+
+        it('should be able to preserve the version', function(done) {
+          presets.updateVersion.cargo({}, this.cwd.name, '1.2.3', (error) => {
+            m.chai.expect(error).to.not.exist;
+
+            m.chai.expect(fs.readFileSync(this.cargoToml, 'utf8')).to.equal([
+              '[package]',
+              'name = "foo"',
+              'version = "1.2.3"',
+              '',
+              '[dependencies.bar]',
+              'version = "2.0.0"',
+              ''
+            ].join('\n'));
```

Comment:
Given that the "before-string" and the "after-string" are identical in this particular test-case, would it be worth storing it in a single const-variable and using that in both places?

Reply:
Right, I have not thought of it. Nice idea.